samza-azure/src/main/java/org/apache/samza/checkpoint/azure/AzureCheckpointManager.java

```diff
@@ -0,0 +1,139 @@
+  private CloudTable cloudTable;
+
+  AzureCheckpointManager(AzureConfig azureConfig) {
+    storageConnectionString = azureConfig.getAzureConnect();
+    azureClient = new AzureClient(storageConnectionString);
+  }
+
+  @Override
+  public void start() {
+    try {
+      // Create the table if it doesn't exist.
+      cloudTable = azureClient.getTableClient().getTableReference(CHECKPOINT_MANAGER_TABLE_NAME);
+      cloudTable.createIfNotExists();
+    } catch (URISyntaxException e) {
+      LOG.error("Connection string {} specifies an invalid URI while creating checkpoint table.",
+              storageConnectionString);
+    } catch (StorageException e) {
+      LOG.error("Azure Storage failed when creating a table", e);
+    }
+  }
+
+  @Override
+  public void register(TaskName taskName) {
+    taskNames.add(taskName);
+  }
+
+  @Override
+  public void writeCheckpoint(TaskName taskName, Checkpoint checkpoint) {
+    if (!taskNames.contains(taskName)) {
+      throw new SamzaException("writing checkpoint of unregistered task");
+    }
+
+    TableBatchOperation batchOperation = new TableBatchOperation();
+
+    checkpoint.getOffsets().forEach((ssp, offset) -> {
+        TaskCheckpointEntity taskCheckpoint = new TaskCheckpointEntity(taskName.toString(), ssp.getSystem());
+        taskCheckpoint.setStreamName(ssp.getStream());
+        taskCheckpoint.setPartitionId(ssp.getPartition().getPartitionId());
+        taskCheckpoint.setOffset(offset);
+        batchOperation.insertOrReplace(taskCheckpoint);
+      });
+
+    try {
+      cloudTable.execute(batchOperation);
+    } catch (StorageException e) {
+      LOG.error("Connection string {} specifies an invalid key.", storageConnectionString);
+      throw new AzureException(e);
+    }
+  }
+
+  @Override
+  public Checkpoint readLastCheckpoint(TaskName taskName) {
+    if (!taskNames.contains(taskName)) {
+      throw new SamzaException("writing checkpoint of unregistered task");
+    }
+    String partitionQueryKey = taskName.toString();
+    String partitionFilter = TableQuery.generateFilterCondition(
+            PARTITION_KEY,
+            TableQuery.QueryComparisons.EQUAL,
+            partitionQueryKey);
+
+    TableQuery<TaskCheckpointEntity> partitionQuery = TableQuery.from(TaskCheckpointEntity.class)
+            .where(partitionFilter);
+    Map<SystemStreamPartition, String> offsets = new HashMap<>();
+    for (TaskCheckpointEntity taskCheckpointEntity : cloudTable.execute(partitionQuery)) {
+      String systemName = taskCheckpointEntity.getRowKey();
+      String streamName = taskCheckpointEntity.getStreamName();
+      int partitionId = taskCheckpointEntity.getPartitionId();
+      offsets.put(new SystemStreamPartition(systemName, streamName, new Partition(partitionId)),
+              taskCheckpointEntity.getOffset());
+    }
+    return new Checkpoint(offsets);
+  }
+
+  @Override
+  public void stop() {
+    try {
+      cloudTable.deleteIfExists();
```

Comment:
If we delete the table when `checkpointManager.stop` is called, then does this mean that the persisted checkpoint data is lost?

We want to retain checkpoint data to persist even after the job is stopped, because during a restart we want to load from our previous checkpoint.